# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/GpuDecoderModule.h
# Commit: c5dac46ef11d
# Full Hash: c5dac46ef11d44762d5eb61610e58f4d52aec88d
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - PA: Delete GPUDecoderModule wrapper. r=mattwoodrow
#   
#   Moved support logic into WMFDecoderModule.
#   
#   Depends on D54882
# ==============================================================================

diff -r fca05e774f88 -r c5dac46ef11d dom/media/ipc/GpuDecoderModule.h
--- a/dom/media/ipc/GpuDecoderModule.h	Tue Oct 20 23:26:33 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,47 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-#ifndef include_dom_media_ipc_GpuDecoderModule_h
-#define include_dom_media_ipc_GpuDecoderModule_h
-#include "PlatformDecoderModule.h"
-
-#include "MediaData.h"
-
-namespace mozilla {
-
-// A PDM implementation that creates a RemoteMediaDataDecoder (a
-// MediaDataDecoder) that proxies to a VideoDecoderChild.  The
-// VideoDecoderChild will talk to a VideoDecoderParent running on the
-// GPU process.
-// We currently require a 'wrapped' PDM in order to be able to answer
-// SupportsMimeType and DecoderNeedsConversion. Ideally we'd check these
-// over IPDL using the manager protocol
-class GpuDecoderModule : public PlatformDecoderModule {
- public:
-  explicit GpuDecoderModule(PlatformDecoderModule* aWrapped)
-      : mWrapped(aWrapped) {}
-
-  nsresult Startup() override;
-
-  bool SupportsMimeType(const nsACString& aMimeType,
-                        DecoderDoctorDiagnostics* aDiagnostics) const override;
-  bool Supports(const SupportDecoderParams& aParams,
-                DecoderDoctorDiagnostics* aDiagnostics) const override;
-
-  already_AddRefed<MediaDataDecoder> CreateVideoDecoder(
-      const CreateDecoderParams& aParams) override;
-
-  already_AddRefed<MediaDataDecoder> CreateAudioDecoder(
-      const CreateDecoderParams& aParams) override {
-    return nullptr;
-  }
-
- private:
-  RefPtr<PlatformDecoderModule> mWrapped;
-};
-
-}  // namespace mozilla
-
-#endif  // include_dom_media_ipc_GpuDecoderModule_h